src/java.base/share/classes/sun/nio/ch/NioSocketImpl.java

```diff
@@ -642,7 +643,7 @@ protected void bind(InetAddress host, int port) throws IOException {
             NetHooks.beforeTcpBind(fd, host, port);
             Net.bind(fd, host, port);
             // set the address field to the given host address to keep
-            // compatibility with PlainSocketImpl. When binding to 0.0.0.0
+            // compatibility with SocketImpl. When binding to 0.0.0.0
             // then the actual local address will be ::0 when IPv6 is enabled.
```

Comment:
I don't think "SocketImpl" works here. Maybe replace "keep compatibility with PlainSocketImpl" with "to maintain long standing behavior".

Reply:
Wording replaced as suggested. See 49125e7